### PR TITLE
Bug Fix and Add sth

### PR DESCRIPTION
**Pull request by wxkj99**
A bug in "picture.hpp" has been fixed.
"castle.hpp" has been protected.
Add an example, if it works well, soon it may turn into a header.

### CODE REVIEW

File: render.hpp [hunk 1]
```diff
@@ -25,10 +22,8 @@ void render(BytePicture bitmap,
 	auto f = [width, get_color, &bitmap](Dword begin, Dword end) {
 		for (Dword i = begin; i < end; ++i)
 			for (Dword j = 0; j < width; ++j) {
-				auto c = get_color(i, j);
-				bitmap.data[i][j][0] = get<0>(c);
-				bitmap.data[i][j][1] = get<1>(c);
-				bitmap.data[i][j][2] = get<2>(c);
+				T c = get_color(i, j);
+				bitmap.data[i][j] = c;
```
Comment (alphagocc) [inline render.hpp]: 改成 bitmap.data[i][j]=get_color(i,j)

File: example/circle.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#include "castle.hpp"
+#include <bits/stdc++.h>
```
Comment (alphagocc) [inline example/circle.cpp:2]: Better Not Use bits/stdc++.h

Comment (wxkj99) [reply example/circle.cpp:2]: It is just a example.
I won't change it.

File: render.hpp (hunk 1 above)
Comment (wxkj99) [reply render.hpp]: OK.

File: example/circle.cpp (hunk 2 above)
Comment (alphagocc) [reply example/circle.cpp:2]: Ah,MSVC and Clang does't have this header

Comment (wxkj99) [reply example/circle.cpp:2]: ok

**Pull request merged**